db/helpers/generatorUtils.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * db/helpers/generatorUtils.js
+ *
+ * DB utils for the generator model.
+ */
+'use strict'; // eslint-disable-line strict
+
+const crypto = require('crypto');
+const dbConstants = require('../constants');
+
+/**
+ * Encrypt the given data using passed in secretKey and algorithm
+ * @param  {String} data -  Data to be encrypted
+ * @param  {String} secretKey - Secret key for encryption
+ * @param  {String} algorithm - Name of the encryption algorithm
+ * @returns {String} - encrypted data
+ */
+function encrypt(data, secretKey, algorithm) {
+  const cipher = crypto.createCipher(algorithm, secretKey);
+  let crypted = cipher.update(data, 'utf-8', 'hex');
+  crypted += cipher.final('hex');
+  return crypted;
+} // encrypt
+
+/**
+ * Decrypt encrypted data using passed in secretKey and algorithm
+ * @param  {String} data -  Data to be decrypted
+ * @param  {String} secretKey - Secret key that was used from encryption
+ * @param  {String} algorithm - Name of the encryption algorithm
+ * @returns {String} - encrypted data
+ */
+function decrypt(data, secretKey, algorithm) {
+  const decipher = crypto.createDecipher(algorithm, secretKey);
+  let decrypted = decipher.update(data, 'hex', 'utf-8');
+  decrypted += decipher.final('utf-8');
+  return decrypted;
+} // decrypt
+
+/**
+ * This function fetches the sample generator secret key and the sample
+ * generator encryption algorithm from the glocalConfig model
+ * and returns an object containing the secretKey and algorithm
+ * @param {Model} globalConfigModel - The sequelize Global Config Model
+ * @returns {Object} - containing the secret key and algorithm
+ */
+function getSGEncryptionKeyAndAlgorithm(globalConfigModel) {
```

Comment:
I am not sure why are we passing model around. Is it not safe to assume that key/algo will always be in globalConfigModel?

Reply:
Because the key and algorithm rows are added by the Admin, it is not safe to assume that these rows are always going to be there.

Reply:
Right, but the model will always be there, correct?

Reply:
> its just that we will have to import and pass model (eg in api controller) whenever we wanna use these functions.
utility/helper files should more or less have no dependency on other parts of the code. You already have access to the models in the api layer in api/v1/helpers/nouns